### PR TITLE
update tsconfig, beginning DependenciesCtrl

### PR DESCRIPTION
**Pull request by kiyomih**
WIP, but thought I'd open early PR for this, to get the ball rolling on discussing some things like data contract between FE and BE, and endpoints we have & their formats.

Will add comments throughout wrt these points.

### CODE REVIEW

File: src/rest/DependenciesCtrl.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import restify = require('restify');
+
+export enum DependencyTypes {
```
Comment (kiyomih) [inline src/rest/DependenciesCtrl.ts]: Possible types of dependencies

File: src/rest/DependenciesCtrl.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+     * @returns {Promise<any>}
+     */
+    public async start(): Promise<any> {
+
+        console.log("STARTING UP SERVER: " + this.server.name);
+
+        this.server.use(restify.plugins.acceptParser(this.server.acceptable));
+        this.server.use(restify.plugins.queryParser());
+        this.server.use(restify.plugins.bodyParser());
+
```
Comment (kiyomih) [inline src/rest/DependenciesCtrl.ts]: Any other types of data FE will want to display?

And what does everyone think of the endpoint format `/<graph-type>/<level>/:<startcommit SHA>..<endcommit SHA>`

File: src/rest/DependenciesCtrl.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+     *
+     * Note: we can update return value in promise to something more useful, currently void.
+     * @returns {Promise<any>}
+     */
+    public async stop(): Promise<any> {
+        return this.server.close(() => {
+            console.log("CLOSING SERVER");
+        });
+    }
+
```
Comment (kiyomih) [inline src/rest/DependenciesCtrl.ts]: Below are examples of possible data format we send frontend.

In general, the idea is that it is a 2D matrix of Files/ Classes/ Functions, and each spot in the matrix is filled with the dependency %age reference value.

File: src/rest/DependenciesCtrl.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            data: [
+                [1, 0.2, 0.8],
+                [0.5, 1, 0.3],
+                [0.7, 0.6, 1]
+            ]
+        };
+        res.send(sampleData);
+        return next();
+    }
+
```
Comment (kiyomih) [inline src/rest/DependenciesCtrl.ts]: Similar idea for regular dependencies, but each cell in the matrix would be an array to allow for multiple types of dependencies between Files/Classes/Functions where appropriate.

File: src/rest/DependenciesCtrl.ts (hunk 2 above)
Comment (jyoo980) [reply src/rest/DependenciesCtrl.ts]: Sounds good to me, does restify have support for query parameters? Something like

```
GET /deps/:graphType/:level?start=<startSHA>&end=<endSHA>
```

**Comment by jyoo980** [comment]
Looks good to me so far! Thanks for doing this.

File: src/rest/DependenciesCtrl.ts (hunk 3 above)
Comment (scveloso) [reply src/rest/DependenciesCtrl.ts]: Sounds good to me as long as we specify whether its column depends on row or row depends on column

Comment (daviidli) [reply src/rest/DependenciesCtrl.ts]: Also I think the node sizes need to be sent if we want to show that, like:
`sizes: [10, 20, 30]` or we can normalize them into a percentage?

Comment (kiyomih) [reply src/rest/DependenciesCtrl.ts]: Good idea! will add that in

**Comment by kiyomih** [comment]
Going to merge in so we have the TSConfig file~ will submit another PR with the suggestions here! TY for looking at first pass 🙏🏻

**Pull request merged**